File: VB/FishDataSource/FishDataSource/Fishes.vb
Imports DevExpress.DataAccess.ObjectBinding
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Xml.Serialization

Namespace FishDataSource
    Public Class Fish
        <DisplayName("ID")> _
        Public Property ID() As Integer
        <DisplayName("Category")> _
        Public Property Category() As String
        <DisplayName("Common Name")> _
        Public Property CommonName() As String
        <DisplayName("Notes")> _
        Public Property Notes() As String
        <DisplayName("Scientific Classification")> _
        Public Property ScientificClassification() As ScientificClassification
    End Class
    Public Class ScientificClassification
        <XmlElement("Reference")> _
        Public Property Hyperlink() As String
        Public Property Kingdom() As String
        Public Property Phylum() As String
        <XmlElement("Class"), DisplayName("Class")> _
        Public Property _Class() As String
        Public Property Order() As String
        Public Property Family() As String
        Public Property Genus() As String
        Public Property Species() As String
    End Class

    <DisplayName("FishesSource"), HighlightedClass> _
    Public NotInheritable Class FishesSource

        Private Sub New()
        End Sub


        Private Shared data_Renamed As List(Of Fish)
        <HighlightedMember> _
        Public Shared ReadOnly Property Data() As List(Of Fish)
            Get
                If data_Renamed Is Nothing Then
                    data_Renamed = GetDataSource()
                End If
                Return data_Renamed
            End Get
        End Property
        Private Shared Function GetDataSource() As List(Of Fish)
            Return DataSourceHelper.GetDataSouresFromXml(Of Fish)("FishDataSource.fishes.xml", "Fishes")
        End Function
    End Class
End Namespace

File: CS/FishDataSource/FishDataSource/Fishes.cs
using DevExpress.DataAccess.ObjectBinding;
using System.Collections.Generic;
using System.ComponentModel;
using System.Xml.Serialization;

namespace FishDataSource {
    public class Fish {
        [DisplayName("ID")]
        public int ID { get; set; }
        [DisplayName("Category")]
        public string Category { get; set; }
        [DisplayName("Common Name")]
        public string CommonName { get; set; }
        [DisplayName("Notes")]
        public string Notes { get; set; }
        [DisplayName("Scientific Classification")]
        public ScientificClassification ScientificClassification { get; set; }
    }
    public class ScientificClassification {
        [XmlElement("Reference")]
        public string Hyperlink { get; set; }
        public string Kingdom { get; set; }
        public string Phylum { get; set; }
        [XmlElement("Class")]
        [DisplayName("Class")]
        public string _Class { get; set; }
        public string Order { get; set; }
        public string Family { get; set; }
        public string Genus { get; set; }
        public string Species { get; set; }
    }

    [DisplayName("FishesSource")]
    [HighlightedClass]
    public static class FishesSource {
        static List<Fish> data;
        [HighlightedMember]
        public static List<Fish> Data
        {
            get
            {
                if (data == null)
                    data = GetDataSource();
                return data;
            }
        }
        static List<Fish> GetDataSource() {
            return DataSourceHelper.GetDataSouresFromXml<Fish>("FishDataSource.fishes.xml", "Fishes");
        }
    }
}
